src/main/java/com/minecolonies/coremod/entity/pathfinding/WalkToProxy.java

```diff
@@ -254,6 +284,48 @@ else if (targetY <= levelDepth && workerY > levelDepth)
             //If he is on the same Y level as his target and both underground, don't use a proxy. Just don't.
             else if (targetY <= levelDepth)
             {
+                double closestNode = Double.MAX_VALUE;
+                Node lastNode = null;
+                for(Map.Entry<Point2D, Node> node : level.getNodes().entrySet())
+                {
+                    double distanceToNode = node.getKey().distance(worker.getPosition().getX(), worker.getPosition().getZ());
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Local variable 'distanceToNode' could be declared final [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)